### PR TITLE
Package the jar analyzers

### LINKED ISSUE #5283: Package the jar analyzers

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5283

### CODE REVIEW

File: build/CopyDependencies/CopyDependencies.targets [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  </PropertyGroup>
+  
+  
+  <Target Name="ClearDirs">
+    <RemoveDir Directories="$(EmbeddedJarDir)" />
+  </Target>
+
+  <Target Name="CopyJars">
+    <ItemGroup>
+      <SourceJars Include="$(JarDownloadDir)\sonar-cfamily-plugin-$(EmbeddedSonarCFamilyAnalyzerVersion).jar" />
```
Comment (georgii-borovinskikh-sonarsource) [inline build/CopyDependencies/CopyDependencies.targets:21]: Could we somehow centralize the list of dependencies between this and DownloadDependencies task? Currently, if we ever add another analyzer, we would need to modify these 2 targets + EmbeddedSonarAnalyzer.props file

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 2]
```diff
@@ -286,7 +286,7 @@
   
   <ItemGroup>
     <Content Include="DownloadedJars\*.jar">
-      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Integration.Vsix.csproj]: or rather with this

File: build/CopyDependencies/CopyDependencies.targets (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply build/CopyDependencies/CopyDependencies.targets:21]: I'll add another ticket for this and we can tackle it if we have time in this sprint or we can handle it in a hardening sprint.

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 3]
```diff
@@ -280,7 +280,13 @@
       <IncludeInVSIX>true</IncludeInVSIX>
       <Link>THIRD-PARTY-NOTICES</Link>
     </Content>
+  </ItemGroup>
 
+  <ItemGroup Label="Sloop Dependencies">
+    <Content Include="$(MSBuildThisFileDirectory)$(JarsFolderName)\*.jar">
+      <IncludeInVSIX>True</IncludeInVSIX>
+      <VSIXSubPath>$(JarsFolderName)</VSIXSubPath>
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/Integration.Vsix.csproj:288]: Is this somehow related? https://github.com/SonarSource/sonarlint-visualstudio/blob/3400f8af51d6f9b44cdf34196ae5ecbe57423ba2/src/Integration.Vsix/SonarLintIntegrationPackage.cs#L47

File: src/Integration.Vsix/packages.lock.json [hunk 4]
```diff
@@ -1533,6 +1533,9 @@
           "Microsoft.VisualStudio.Interop": "17.0.31902.203"
         }
       },
+      "copydependencies": {
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/packages.lock.json:1536]: Why is it lowercase?

File: build/CopyDependencies/CopyDependencies.targets (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [inline build/CopyDependencies/CopyDependencies.targets:21]: Will this fail if it can't find any of those files?

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

File: src/Integration.Vsix/packages.lock.json (hunk 4 above)
Comment (ugras-ergun-sonarsource) [reply src/Integration.Vsix/packages.lock.json:1536]: no idea it's auto generated.

File: src/Integration.Vsix/Integration.Vsix.csproj (hunk 3 above)
Comment (ugras-ergun-sonarsource) [reply src/Integration.Vsix/Integration.Vsix.csproj:288]: I don't see how these can be related. The one you referred is about loading assemblies, we are not interested in that here.

File: build/CopyDependencies/CopyDependencies.targets (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply build/CopyDependencies/CopyDependencies.targets:21]: Yes, and I believe it's desired behaviour.

File: src/Integration.Vsix/packages.lock.json (hunk 4 above)
Comment (ugras-ergun-sonarsource) [reply src/Integration.Vsix/packages.lock.json:1536]: Some other stuff is also lowercase (e.g. "sonarqube.client"). Since it doesn't affect anything other than cosmetics on an auto-generated file I don't consider it as a problem.
